pkg/extension/sumologicextension/extension.go

```diff
@@ -286,10 +286,19 @@ func (se *SumologicExtension) registerCollector(ctx context.Context, collectorNa
 			zap.Int("status_code", res.StatusCode),
 			zap.String("response", buff.String()),
 		)
-		return CollectorCredentials{}, fmt.Errorf(
-			"failed to register the collector, got HTTP status code: %d",
-			res.StatusCode,
-		)
+
+		// Return unrecoverable error for 4xx status codes except 429
+		if res.StatusCode >= 400 && res.StatusCode < 500 && res.StatusCode != 429 {
+			return CollectorCredentials{}, backoff.Permanent(fmt.Errorf(
+				"failed to register the collector, got HTTP status code: %d",
```

Comment:
Not strictly related to this very PR, but aren't we losing the original error's message and only passing the response code here? Wouldn't the original error's message potentially contain additional info for the user, helping them fix the problem?

Reply:
Response is logged above
```
		se.logger.Debug("Collector registration failed",
			zap.Int("status_code", res.StatusCode),
			zap.String("response", buff.String()),
		)
```